src/main/java/jeo/parser/Parser.java

```diff
@@ -0,0 +1,182 @@
+        } else if (str.toLowerCase().startsWith("event")) {
+            hm.put("command", "event");
+            str = str.substring(5).trim();
+            if (!str.toLowerCase().contains("/from") || !str.toLowerCase().contains("/to")) {
+                throw new JeoException(str.isEmpty()
+                        ? "Please enter a task description."
+                        : "Please follow the format: event <description> "
+                        + "/from <yyyy-MM-dd HH:mm> /to <yyyy-MM-dd HH:mm>");
+            }
+            String[] arr = parseSubstringTasks(str, "event1");
+            String desc = arr[0];
+            str = arr[1];
+            arr = parseSubstringTasks(str, "event2");
+            String from = arr[0];
+            String to = arr[1];
+            if (desc.isEmpty()) {
+                throw new JeoException("Please enter a task description.");
+            }
+            if (from.isEmpty() && to.isEmpty()) {
+                throw new JeoException("Please enter a date-time in the format "
+                        + "\"yyyy-MM-dd HH:mm\" after \"/from\" and \"/to\".");
+            }
+            if (from.isEmpty()) {
+                throw new JeoException("Please enter a date-time in the format "
+                        + "\"yyyy-MM-dd HH:mm\" after \"/from\".");
+            }
+            if (to.isEmpty()) {
+                throw new JeoException("Please enter a date-time in the format "
+                        + "\"yyyy-MM-dd HH:mm\" after \"/to\".");
+            }
+            hm.put("description", desc);
+            hm.put("from", from);
+            hm.put("to", to);
+        } else if (str.toLowerCase().startsWith("mark")) {
+            hm.put("command", "mark");
+            int i = parseSubStringActions(str, 4);
+            hm.put("index", Integer.toString(i - 1));
+        } else if (str.toLowerCase().startsWith("unmark")) {
+            hm.put("command", "unmark");
+            int i = parseSubStringActions(str, 6);
+            hm.put("index", Integer.toString(i - 1));
+        } else if (str.toLowerCase().startsWith("delete")) {
+            hm.put("command", "delete");
+            int i = parseSubStringActions(str, 6);
+            hm.put("index", Integer.toString(i - 1));
+        } else if (str.toLowerCase().startsWith("due")) {
+            hm.put("command", "due");
+            String by = str.substring(3).trim();
+            if (by.isEmpty()) {
+                throw new JeoException("Please enter a date in the format: \"yyyy-MM-dd\".");
+            }
+            hm.put("by", by);
+        } else if (str.toLowerCase().startsWith("find")) {
+            hm.put("command", "find");
+            String key = str.substring(4).trim();
+            if (key.isEmpty()) {
+                throw new JeoException("Please enter a keyword.");
+            }
+            hm.put("key", key);
+        } else {
+            hm.put("command", "");
+        }
+        return hm;
```

Comment:
Similar to another comment on the same block of if-else code, with such a long block the readability is slightly compromised (and it is slightly harder to debug), so you may consider editing the code to cater to that (switch cases etc.).